testers/testers/markus_tester.py

```diff
@@ -218,49 +254,38 @@ def before_tester_run(self):
         """
         pass
 
-    def before_test_run(self, test):
-        """
-        Callback invoked before running a test.
-        Use this for test initialization steps that can fail, rather than using test_class.__init__().
-        :param test: The test after initialization.
-        """
-        pass
-
-    def after_successful_test_run(self, test):
-        """
-        Callback invoked after successfully running a test.
-        Use this to access test data in the tester. Don't use this for test cleanup steps, use test_class.run() instead.
-        :param test: The test after execution.
-        """
-        pass
-
     def after_tester_run(self):
         """
         Callback invoked after running this tester, including in case of exceptions.
         Use this for tester cleanup steps that should always be executed, regardless of errors.
         """
         pass
 
+    @staticmethod
+    def run_decorator(run_func):
+        @wraps(run_func)
+        def run_func_wrapper(self, *args, **kwargs):
+            try:
+                self.before_tester_run()
+                return run_func(self, *args, **kwargs)
+            except Exception as e:
+                print(MarkusTester.error_all(message=str(e)), flush=True)
+            finally:
+                self.after_tester_run()
+        return run_func_wrapper
+
+    @staticmethod
+    @contextmanager
+    def open_feedback(filename, mode='w'):
+        if filename:
+            try:
+                feedback_open = open(filename, mode)
+                yield feedback_open
+            finally:
+                feedback_open.close()
```

Comment:
I think you should check for feedback_open to not be None here, in case the open fails?